sql/core/src/test/resources/sql-tests/results/group-by-filter.sql.out

```diff
@@ -0,0 +1,332 @@
+FROM (SELECT 1 AS a, 2 AS b, 3 AS c) GROUP BY a
+-- !query 12 schema
+struct<count(DISTINCT b):bigint,count(DISTINCT b, c):bigint>
+-- !query 12 output
+1	1
+
+
+-- !query 13
+SELECT a AS k, COUNT(b) FILTER (WHERE b = 1 OR b = 2) FROM testData GROUP BY k
+-- !query 13 schema
+struct<k:int,count(b):bigint>
+-- !query 13 output
+1	2
+2	2
+3	2
+NULL	1
+
+
+-- !query 14
+SELECT a AS k, COUNT(b) FILTER (WHERE NOT b < 0) FROM testData GROUP BY k HAVING k > 1
+-- !query 14 schema
+struct<k:int,count(b):bigint>
+-- !query 14 output
+2	2
+3	2
+
+
+-- !query 15
+SELECT COUNT(b) FILTER (WHERE a > 0) AS k FROM testData GROUP BY k
+-- !query 15 schema
+struct<>
+-- !query 15 output
+org.apache.spark.sql.AnalysisException
+aggregate functions are not allowed in GROUP BY, but found count(testdata.`b`);
+
+
+-- !query 16
+SELECT a AS k, COUNT(b) FILTER (WHERE b > 0) FROM testData GROUP BY k
+-- !query 16 schema
+struct<k:int,count(b):bigint>
+-- !query 16 output
+1	2
+2	2
+3	2
+NULL	1
+
+
+-- !query 17
+SELECT a, COUNT(1) FILTER (WHERE b > 1) FROM testData WHERE false GROUP BY a
+-- !query 17 schema
+struct<a:int,count(1):bigint>
+-- !query 17 output
+
+
+
+-- !query 18
+SELECT COUNT(1) FILTER (WHERE b = 2) FROM testData WHERE false
+-- !query 18 schema
+struct<count(1):bigint>
+-- !query 18 output
+0
+
+
+-- !query 19
+SELECT 1 FROM (SELECT COUNT(1) FILTER (WHERE a >= 3 OR b <= 1) FROM testData WHERE false) t
+-- !query 19 schema
+struct<1:int>
+-- !query 19 output
+1
+
+
+-- !query 20
+CREATE TEMPORARY VIEW EMP AS SELECT * FROM VALUES
+  (100, "emp 1", date "2005-01-01", 100.00D, 10),
+  (100, "emp 1", date "2005-01-01", 100.00D, 10),
+  (200, "emp 2", date "2003-01-01", 200.00D, 10),
+  (300, "emp 3", date "2002-01-01", 300.00D, 20),
+  (400, "emp 4", date "2005-01-01", 400.00D, 30),
+  (500, "emp 5", date "2001-01-01", 400.00D, NULL),
+  (600, "emp 6 - no dept", date "2001-01-01", 400.00D, 100),
+  (700, "emp 7", date "2010-01-01", 400.00D, 100),
+  (800, "emp 8", date "2016-01-01", 150.00D, 70)
+AS EMP(id, emp_name, hiredate, salary, dept_id)
+-- !query 20 schema
+struct<>
+-- !query 20 output
+
+
+
+-- !query 21
+CREATE TEMPORARY VIEW DEPT AS SELECT * FROM VALUES
+  (10, "dept 1", "CA"),
+  (20, "dept 2", "NY"),
+  (30, "dept 3", "TX"),
+  (40, "dept 4 - unassigned", "OR"),
+  (50, "dept 5 - unassigned", "NJ"),
+  (70, "dept 7", "FL")
+AS DEPT(dept_id, dept_name, state)
+-- !query 21 schema
+struct<>
+-- !query 21 output
+
+
+
+-- !query 22
+SELECT emp.dept_id,
+       avg(salary),
+       avg(salary) FILTER (WHERE EXISTS (SELECT state
+               FROM dept
+               WHERE dept.dept_id = emp.dept_id))
+FROM emp
+GROUP BY dept_id
+-- !query 22 schema
+struct<>
+-- !query 22 output
+org.apache.spark.sql.AnalysisException
+IN/EXISTS predicate sub-queries can only be used in Filter/Join and a few commands: Aggregate [dept_id#x], [dept_id#x, avg(salary#x) AS avg(salary)#x, avg(salary#x) AS avg(salary)#x]
```

Comment:
Is this expected?

Reply:
I don't know. it seems is forbidden or other reason.

Reply:
Can you check the PgSQL behaviour? If it supports this query, we should do.

Reply:
```
[SQL]SELECT COMPANY.DEP_ID,
       avg(SALARY),
       avg(SALARY) FILTER (WHERE EXISTS (SELECT ID
               FROM DEPARTMENT
               WHERE DEPARTMENT.ID = COMPANY.DEP_ID))
FROM COMPANY;

[Err] ERROR:  column "company.dep_id" must appear in the GROUP BY clause or be used in an aggregate function
LINE 1: SELECT COMPANY.DEP_ID,
               ^
```

Reply:
I tried and the it seemed that the query passed in pg;
```
postgres=# SELECT emp.dept_id, avg(salary), avg(salary) FILTER (WHERE EXISTS (SELECT state FROM dept WHERE dept.dept_id = emp.dept_id)) FROM emp GROUP BY dept_id;
 dept_id |         avg          |         avg          
---------+----------------------+----------------------
         | 400.0000000000000000 |                     
      10 | 133.3333333333333333 | 133.3333333333333333
     100 | 400.0000000000000000 |                     
      30 | 400.0000000000000000 | 400.0000000000000000
      70 | 150.0000000000000000 | 150.0000000000000000
      20 | 300.0000000000000000 | 300.0000000000000000
(6 rows)
```
Do I miss something?

Reply:
```
SELECT COMPANY.DEP_ID,
       avg(SALARY),
       avg(SALARY) FILTER (WHERE EXISTS (SELECT ID
               FROM DEPARTMENT
               WHERE DEPARTMENT.ID = COMPANY.DEP_ID))
FROM COMPANY GROUP BY DEP_ID;
```
I'm sorry, I lost the group by clause.
PgSQL supports this query.

Reply:
If so, we should support this, too.

Reply:
It seems difficult to implement. 
I references the implement that IN/EXISTS predicate sub-queries used in Filter, it converts the subquery to semi/anti join. But we cannot do the same.
Do you have a good idea?

Reply:
The ohter dbms-like systems support this query, oracle, mysql, presto, ...? Can you check?

Reply:
I just checked oracle, mysql, presto, hive, Redshift, Vertica, DB2, all of them not support this query.

Reply:
Can you show us error messages of part of them(e.g., oracle and db2)? What's a kind of errors?

Reply:
The error message of MySQL is:
```
[SQL]SELECT emp.dept_id,
       avg(salary),
       avg(salary) FILTER (WHERE EXISTS (SELECT state
               FROM dept
               WHERE id = emp.dept_id))
FROM emp
GROUP BY dept_id;
[Err] 1064 - You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '(WHERE EXISTS (SELECT state
               FROM dept
               WHERE id =' at line 3
```

Reply:
How about Oracle? It has more meaningful errors like "FILTER syntax does not support subqueries, brabrabra"?

Reply:
The error message of Oracle is:
```
SELECT TEST."emp".dept_id, avg(salary), avg(salary) FILTER (WHERE EXISTS (SELECT state FROM TEST."dept" WHERE id = TEST."emp".dept_id))
FROM TEST."emp" GROUP BY dept_id
> ORA-00923: FROM keyword not found where expected
  
> Time: 0.026s
```

Reply:
Ah, I see. Thanks. Then, yes, I personally think its ok not to support subqueries in this pr. Can you throw an exception with an explicit error message in `CheckAnalysis`?

Reply:
Ah, I see. Thanks. Then, yes, I personally think its ok not to support subqueries in this pr. Can you throw an exception with an explicit error message in `CheckAnalysis`?  cc: @gatorsmile @cloud-fan